src/bridge/BridgeTransport.cpp

```diff
@@ -0,0 +1,138 @@
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+#include "BridgeTransport.hpp"
+
+void BridgeTransport::Start() {
+    thread_ = std::make_unique<std::thread>(&BridgeTransport::RunThread, this);
+}
+
+void BridgeTransport::Stop() {
+    if (!thread_ || !thread_->joinable()) return;
+    StopAsync();
+    logger_->Log("stopping");
+    thread_->join();
+    thread_.reset();
+}
+
+void BridgeTransport::StopAsync() {
+    if (!stop_signal_handle_ || stop_signal_handle_->closing()) return;
+    stop_signal_handle_->send();
+}
+
+void BridgeTransport::RunThread() {
+    logger_->Log("thread started");
+    loop_ = uvw::loop::create();
+    stop_signal_handle_ = GetLoop()->resource<uvw::async_handle>();
+    write_signal_handle_ = GetLoop()->resource<uvw::async_handle>();
+
+    stop_signal_handle_->on<uvw::async_event>([this](const uvw::async_event&, uvw::async_handle& handle) {
+        logger_->Log("closing handles");
+        CloseConnectionHandles();
+        write_signal_handle_->close();
+        stop_signal_handle_->close();
+    });
+    
+    write_signal_handle_->on<uvw::async_event>([this](const uvw::async_event&, uvw::async_handle& handle) {
+        SendWrites();
+    });
+    
+    CreateConnection();
+    GetLoop()->run();
+    GetLoop()->close();
+    logger_->Log("thread exited");
+}
+
+void BridgeTransport::ResetBuffers() {
+    recv_buf_.Clear();
+    send_buf_.Clear();
+}
+
+void BridgeTransport::OnRecv(const uvw::data_event& event) {
+    if (!recv_buf_.Push(event.data.get(), event.length)) {
+        logger_->Log("recv_buf_.Push(%i) failed", event.length);
+        ResetConnection();
+        return;
+    }
+
+    size_t available;
+    while (available = recv_buf_.BytesAvailable()) {
+        if (available < 4) return;
+
+        char len_buf[4];
+        recv_buf_.Peek(len_buf, 4);
+        uint32_t size = LE32_TO_NATIVE(*reinterpret_cast<uint32_t*>(len_buf));
```

Comment:
same nit as the write case, but in reverse. Feel free to write a macro for this.
```suggestion
        uint32_t size = (len_buf[0] << 0) | (len_buf[1] << 8) | (len_buf[2] << 16) | (len_buf[3] << 24);
```